src/tools/tidy/src/bins.rs

```diff
@@ -96,9 +96,25 @@ mod os_impl {
 
     #[cfg(unix)]
     pub fn check(path: &Path, bad: &mut bool) {
+        const ALLOWED: &[&str] = &["configure"];
+
         crate::walk_no_read(
             path,
-            &mut |path| crate::filter_dirs(path) || path.ends_with("src/etc"),
+            &mut |path| {
+                crate::filter_dirs(path)
+                    || path.ends_with("src/etc")
+                    // This is a list of directories that we almost certainly
+                    // don't need to walk. A future PR will likely want to
+                    // remove these in favor of crate::walk_no_read using git
+                    // ls-files to discover the paths we should check, which
+                    // would naturally ignore all of these directories. It's
+                    // also likely faster than walking the directory tree
+                    // directly (since git is just reading from a couple files
+                    // to produce the results).
+                    || path.ends_with("target")
+                    || path.ends_with("build")
```

Comment:
Hmm, why special-case these here instead of adding them to `filter_dirs`? `src/etc` makes sense, but the rest seem like other passes will want to ignore them too.

Reply:
I'd rather not go through all the other steps to make sure there's no subtle interaction with these to fix this problem; git ls-files will more thoroughly solve this and avoid needing to think of whether we have a src/test/ui/build or whatever. src/etc is pre-existing, not introduced by this PR.

Reply:
👍